src/NetscriptFunctions.ts

```diff
@@ -1347,13 +1363,18 @@ export const ns: InternalAPI<NSFull> = {
     helpers.log(ctx, () => `Could not find server ${hostname} as a purchased server. This is a bug. Report to dev.`);
     return false;
   },
-  getPurchasedServers: () => (): string[] => {
-    const res: string[] = [];
-    Player.purchasedServers.forEach(function (hostname) {
-      res.push(hostname);
-    });
-    return res;
-  },
+  getPurchasedServers:
+    (ctx) =>
+    (_returnOpts): string[] => {
+      const returnOpts = helpers.hostReturnOptions(_returnOpts);
+      const res: string[] = [];
+      Player.purchasedServers.forEach(function (hostname) {
+        const server = helpers.getServer(ctx, hostname);
+        const id = helpers.returnServerID(server, returnOpts);
+        res.push(id);
+      });
```

Comment:
My vendetta is showing XD (map, reduce, etc. are fine, but `forEach` is not actually part of functional programming and has a bunch of hidden gotchas.)
```suggestion
      for (const hostname of Player.purchasedServers) {
        const server = helpers.getServer(ctx, hostname);
        const id = helpers.returnServerID(server, returnOpts);
        res.push(id);
      }
```

Reply:
In my defense, I think I simply added logic to an already existing `.forEach()` xD
I have no qualms about being the one to implement the change, though, so I'll include it in my next push.